### PR TITLE
JETUT-111 Add example for the open id connect authentication mechanism

### PR DESCRIPTION
**Pull request by arjantijms**

### CODE REVIEW

File: focused/security/restOpenIdConnectAuth/src/test/java/jakartaee/examples/focused/security/restopenidconnectauth/RestOpenIdConnectAuthIT.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        page.getElementById("j_password")
+            .setAttribute("value", "password");
+
+        // Submit login form
+        HtmlPage confirmationPage = page.getElementByName("submit")
+                                        .click();
+
+        System.out.println(confirmationPage.asXml());
+
+        // Confirm
```
Comment (kito99) [inline focused/security/restOpenIdConnectAuth/src/test/java/jakartaee/examples/focused/security/restopenidconnectauth/RestOpenIdConnectAuthIT.java:63]: You may want to explain that the confirmation page has an "authorize" button which then goes to the originally requested resource.

**Review by kito99 — APPROVED** [review]

File: focused/security/restOpenIdConnectAuth/pom.xml [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+    <build>
+        <finalName>openid-client</finalName>
+        <plugins>
+            <plugin>
+                
+                <!-- 
+                    Unpack and install Tomcat + Mitre 
+                    
+                    Mitre is a Spring based OpenID Connect Server that best runs on a javax based Tomcat.
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply focused/security/restOpenIdConnectAuth/pom.xml:55]: That would be Jakarta EE 8 based. 9 already has the new namespace ;)

File: focused/security/restOpenIdConnectAuth/src/main/java/jakartaee/examples/focused/security/restopenidconnectauth/ApplicationConfig.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+package jakartaee.examples.focused.security.restopenidconnectauth;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.security.enterprise.authentication.mechanism.http.OpenIdAuthenticationMechanismDefinition;
+import jakarta.ws.rs.ApplicationPath;
+import jakarta.ws.rs.core.Application;
+
+@OpenIdAuthenticationMechanismDefinition(
+    providerURI = "https://localhost:8443/openid-connect-server-webapp",
+    clientId = "client",
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply focused/security/restOpenIdConnectAuth/src/main/java/jakartaee/examples/focused/security/restopenidconnectauth/ApplicationConfig.java:22]: They are explained in the tutorial text, but I'll briefly mention something there too.

File: focused/security/restOpenIdConnectAuth/pom.xml [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+                                <echo level="info">Replacing in ${tomcat.dir}</echo>
+                                
+                                <!-- Configure Mitre to let it know its running on HTTPS  -->
+                                <replace token="http://localhost:8080" value="https://localhost:8443" dir="${tomcat.dir}/webapps/openid-connect-server-webapp/WEB-INF" summary="yes">
+                                    <include name="server-config.xml" />
+                                </replace>
+                                
+                                <!-- Configure Mitre to let it know where the Open ID callback needs to go to -->
+                                <replace token="http://localhost/" value="http://localhost:8080/openid-client/Callback" dir="${tomcat.dir}/webapps/openid-connect-server-webapp/WEB-INF/classes/db/hsql" summary="yes">
+                                    <include name="clients.sql" />
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply focused/security/restOpenIdConnectAuth/pom.xml:154]: It's `${tomcat.dir}/webapps/openid-connect-server-webapp/WEB-INF/classes/db/hsql`, which is the location where the Mitre server is unpacked.

**Pull request merged**